src/cascade/executor/construct_model.py

```diff
@@ -147,16 +148,65 @@ def constraint_from_rectangular_data(rate_var, default_age_time):
 
 
 def smooth_grid_from_smoothing_form(default_age_time, single_age_time, smooth):
+    """
+    Create a new SmoothGrid from the settings in EpiViz-AT at the Smoothing
+    level.
+
+    Args:
+        default_age_time (List[ages, times]): Two members, the ages and the time.
+        single_age_time (List[float]): Two members, an age and a time.
+        smooth (cascade.input_data.configuration.form.Smoothing): The form element.
+    Returns:
+        SmoothGrid: A new smooth grid.
+    """
     ages, times = construct_grid_ages_times(default_age_time, single_age_time, smooth)
     rate_grid = SmoothGrid(ages=ages, times=times)
     for kind in ["value", "dage", "dtime"]:
-        if not smooth.default.is_field_unset(kind):
+        if not smooth.is_field_unset("default") and not smooth.default.is_field_unset(kind):
             getattr(rate_grid, kind)[:, :] = getattr(smooth.default, kind).prior_object
-        else:
-            pass  # An unset prior should be unused (dage for one age, dtime for one time)
+        if not smooth.is_field_unset("mulstd") and not smooth.mulstd.is_field_unset(kind):
+            getattr(rate_grid, kind).mulstd_prior = getattr(smooth.mulstd, kind).prior_object
+    if not smooth.is_field_unset("detail"):
+        for smoothing_prior in smooth.detail:
+            for a, t in matching_knots(rate_grid, smoothing_prior):
+                getattr(rate_grid, smoothing_prior.prior_type)[a, t] = smoothing_prior.prior_object
     return rate_grid
 
 
+def matching_knots(rate_grid, smoothing_prior):
+    """
+    Get lower and upper out of the smoothing prior. This uses the age, time,
+    and "born" lower and upper bounds to return
+    ages and times in the grid that are within those bounds.
+    The goal is to apply a prior selectively to those knots.
+
+    Args:
+        smoothing_prior (cascade.input_data.configuration.form.SmoothingPrior):
+            A single smoothing prior.
+
+    Returns:
+        Iterator over (a, t) that match. Can be nothing.
+    """
+    extents = dict()
+    for extent in ["age", "time", "born"]:
+        extents[extent] = np.zeros(2, dtype=np.float)
+        for sidx, side, default_extent in [(0, "lower", -inf), (1, "upper", inf)]:
```

Comment:
I stumbled over `sidx`. I think `s_idx` or `side_idx` would have been clear to me.

Reply:
That helps.